openapi-extension/src/main/java/io/ballerina/openapi/extension/doc/gen/AbstractOpenApiDocGenerator.java

```diff
@@ -157,8 +155,10 @@ private void generateOpenApiDoc(OpenApiDocConfig config, SyntaxNodeAnalysisConte
                 config.getServiceNode(), listenerNodes, config.getSemanticModel(), targetFile, null);
         Optional<OpenAPI> openApiOpt = oasResult.getOpenAPI();
         if (!oasResult.getDiagnostics().isEmpty() || openApiOpt.isEmpty()) {
-            OpenApiDiagnosticCode errorCode = OpenApiDiagnosticCode.OPENAPI_107;
-            updateCompilerContext(context, location, errorCode);
+            if (embed) {
```

Comment:
Merge condition in L157 and L158

Reply:
Refactor the logic with [5b0b7e4a20209b7e3274b100d2777543d694827e](https://github.com/ballerina-platform/openapi-tools/pull/1108/commits/5b0b7e4a20209b7e3274b100d2777543d694827e)